src/sentry/static/sentry/app/components/charts/baseChart.jsx

```diff
@@ -0,0 +1,98 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import ReactEchartsCore from 'echarts-for-react/lib/core';
+import 'zrender/lib/svg/svg';
+import echarts from 'echarts/lib/echarts';
+
+import theme from 'app/utils/theme';
+
+class BaseChart extends React.Component {
+  static propTypes = {
+    // see: https://ecomfe.github.io/echarts-doc/public/en/option.html
+    options: PropTypes.object,
+
+    // Chart height
+    height: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
+    // Chart width
+    width: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
+    renderer: PropTypes.oneOf(['canvas', 'svg']),
```

Comment:
Would we ever want canvas? Perhaps we shouldn't expose this prop at all

Reply:
I think when you have lots of datapoints you'll want to use `canvas` for performance reasons, I'll add a note in comments for this.